### PR TITLE
Migrate Mixer MCP backend to the new Galley metadata code.

### PR DESCRIPTION
**Pull request by ozevren**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ozevren** [comment]
policy/telemetry folks. Could you take a quick look? This moves Mixer's dependency on config metadata from the old implementation to the new one.

File: galley/pkg/config/schema/kindmapping.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+// Mapping between MCP collections and and CRD Kinds.
+type Mapping struct {
+	// Bidirectional mapping of collections & kinds for non-legacy resources.
+	KindsToCollections map[string]string
+	CollectionsToKinds map[string]string
+}
+
+// ConstructKindMapping constructs a Mapping of kinds to collections. This is mainly used for Mixer v1, and should not
+// be depended on generally
```
Comment (sushicw) [inline galley/pkg/config/schema/kindmapping.go]: Is that also true for CollectionsToKinds?

**Comment by ozevren** [comment]
policy/telemetry folks. Can I get a review, before I turn into a pumpkin?

**Comment by ozevren** [comment]
@geeknoid Do you still have super powers to merge this? This, in effect, the only dependency left to the old metadata model.

File: mixer/pkg/config/mcp/backend.go [hunk 2]
```diff
@@ -175,8 +178,8 @@ func (b *backend) Init(kinds []string) error {
 			requiredFiles = requiredFiles[1:]
 		}
 
-		watcher, err := creds.WatchFiles(ctx.Done(), b.credOptions)
-		if err != nil {
+		watcher, er := creds.WatchFiles(ctx.Done(), b.credOptions)
+		if er != nil {
 			cancel()
 			return err
```
Comment (douglas-reid) [inline mixer/pkg/config/mcp/backend.go]: `return er` ?

Comment (ozevren) [reply mixer/pkg/config/mcp/backend.go]: good catch thanks.

**Comment by ozevren** [comment]
@geeknoid There seems to be an unrelated codegen + mixer problem here

**Comment by geeknoid** [comment]
There's a problem with sporadic crashes in protoc. You just need to retry the gencheck job until it passes. Hopefully, this will be fixed next time we upgrade protoc.

**Pull request merged**